doc/voq/lag_hld.md

```diff
@@ -0,0 +1,427 @@
+*  The "System LAG Table" has an attribute called the "system_lag_id".
+*  Two new tables are defined and are a part of the Centralized Database CHASSIS_APP_DB. They are - "System LAG Table" and "System LAG Membership Table"
+*  The SONiC network stack instance associated with an asic updates the entries in these tables for LAG and LAG Members that are controlled by that asic.
+*  All of the asic SONiC instances subscribe to these two tables and receive updates for remote LAG and LAG membership changes. These updates are programmed into the asic via the SAI APIs.
+Please note that the only per asic SONiC components that are aware of a remote LAG are SWSS and SYNCD. Other components (like teamd for example) are only aware of local LAG.
+
+## 3 Design Details
+
+### 3.1 Configuration
+
+* The **LAG configuration** in VOQ chassis is same as is done in non-VOQ chassis system. No changes are required. LAG expected to be configured on local ports only. No requirement to make system port specific LAG configuration. 
+
+### 3.2 Modules Design
+
+#### teamd: teamsyncd
+
+teamsyncd will act as if its single asic sonic system and on local ports LAG management. So, there is no changes required here. 
+
+#### orchagent: portsorch
+**Local LAG**: The system_lag_id is a locally generated system wide unqiue id for each LAG. After creating local LAG, the PortChannel information is synced to SYSTEM_LAG_TABLE in centralized database CHASSIS_APP_DB. The lag members of the local LAG use local port ids. Any member addition/removal members to a local LAG is synced to SYSTEM_LAG_MEMBER_TABLE in centralized database CHASSIS_DB. While syncing lag members, unique system lag name and system port alias of the local port members are used as the key. Unique system lag name is generated using local Portchannel name and switch id in the format shown below
+```
+Switch<local switch_id>-<local PortChannel name>
+```
+
+**Remote LAG**: To create LAG corresponding to the remote LAG, the system_lag_id will be from corresponding entry from SYSTEM_LAG_TABLE from CHASSIS_APP_DB. The Lag members for a remote will be specified using the OID of the corresponding system port objects.
+
+**System LAG ID**: LAG creation needs unique id across system to be allocated and passed to SAI. Orchagent can do this using switch_id. SAI header has 32 bit number for LAG_ID. First 8 bit will be encoded with switch_id and rest 24 bit will have locally allocated id. This makes LAG id as unique number across system.
+
+```
+   
++--------+--------+--------+--------+
+|          SYSTEM LAG ID            |
++--------+--------+--------+--------+
+| Switch |          LAG ID          |
+|   ID   |                          |
++--------+--------+--------+--------+
+
+```
+
+Changes in orchagent/portsorch include: 
+ * Assign unique system_lag_id across system by using switch_id. The encoding scheme is described as above. 
```

Comment:
When the lag id is configured, make a validation system at the configuration to make sure that it is unique across system. Proposal: take the chassis app db system lag info and populate the local state db and check with this local state db lag id info during configuration.

Reply:
Instead of configuring system_lag_id, acquire it from global redis (with atomic acition insured) so that all asics will get unique available id. By this we can avoid availability validation and race conditions.